versioned/gc/iceberg-actions/src/main/java/org/projectnessie/versioned/gc/actions/GcActions.java

```diff
@@ -123,7 +123,8 @@ private SparkSession spark() {
     IdentifyUnreferencedAssets<Contents, IcebergAssetKey> assets = new IdentifyUnreferencedAssets<>(worker.getValueSerializer(),
         assetKeySerializer, assetKeyConverter, new ValueTypeFilter(worker.getValueSerializer()), spark());
     Dataset<IdentifyUnreferencedAssets.UnreferencedItem> unreferencedAssets = assets.identify(unreferencedValues);
-    long currentCount = spark().read().format("iceberg").load(table.toString()).count();
+    Row maxRunId = spark().read().format("iceberg").load(table.toString()).groupBy().max("runid").first();
```

Comment:
I _assume_ that `groupBy().max("runid")` implicitly returns the right (smallest) "runid"?
Does it make sense to share the code in this line with `purgeUnreferencedAssetTable`?

Reply:
> I _assume_ that `groupBy().max("runid")` implicitly returns the right (smallest) "runid"?

not sure what you mean. `groupBy` with no args does a full aggregation and `max` will take the max of the `runid` so we get the largest (eg last run) of `runid`

Reply:
> Does it make sense to share the code in this line with `purgeUnreferencedAssetTable`?

fixed